plugins/modules/inventory_workflow_manager.py

```diff
@@ -2582,22 +2583,37 @@ def update_interface_detail_of_device(self, device_to_update):
                             self.msg = "The action to clear the MAC Address table is only supported for devices with the ACCESS role."
                             self.log(self.msg, "WARNING")
                             self.result['response'] = self.msg
+                            self.result['changed'] = False
                         else:
                             deploy_mode = interface_params.get('deployment_mode', 'Deploy')
-                            self.clear_mac_address(interface_id, deploy_mode, interface_name)
-                            self.check_return_status()
+                            self.clear_mac_address(interface_id, deploy_mode, interface_name).check_return_status()
 
                     temp_params = {
                         'description': interface_params.get('description', ''),
                         'adminStatus': interface_params.get('admin_status'),
                         'voiceVlanId': interface_params.get('voice_vlan_id'),
                         'vlanId': interface_params.get('vlan_id')
                     }
+                    # import epdb;
+                    # epdb.serve(port=8888)
                     payload_params = {}
                     for key, value in temp_params.items():
                         if value is not None:
                             payload_params[key] = value
 
+                    # Check if interface need update or not here
+                    interface_needs_update = any(key == "voiceVlanId" and str(value) != interface_details['voiceVlan']
+                                                 or str(value) != str(interface_details.get(key)) for key, value in payload_params.items())
+
+                    if not interface_needs_update:
+                        self.status = "success"
+                        self.result['changed'] = False
+                        self.msg = """Interface details for the given interface '{0}' is already updated in the Cisco Catalyst Center for the
```

Comment:
are already updated..

"is" with "are"  plural "details."

Reply:
changed.